loader.c

```diff
@@ -165,17 +168,15 @@ ElfW(Word) get_dynamic_entry(ElfW(Dyn) *dynamic, int field)
 }
 
 void plt_trampoline();
-#if defined(__x86_64__)
-asm(".pushsection \".text\",\"ax\",@progbits\n"
-    "plt_trampoline:\n"
-    "pop %rdi\n" /* Argument 1 */
-    "pop %rsi\n" /* Argument 2 */
-    "call system_plt_resolver\n"
-    "jmp *%rax\n"
-    ".popsection\n");
-#else
-#error "Unsupported architecture"
-#endif
+asm(".pushsection .text,\"ax\",\"progbits\""  "\n"
```

Comment:
More comments are necessary for maintenance purpose since the crucial parts are now moved to `arch/` directory along with extra macros.